client/src/main/java/io/pravega/client/stream/impl/EventStreamReaderImpl.java

```diff
@@ -370,10 +371,18 @@ public Type fetchEvent(EventPointer pointer) throws NoSuchEventException {
         }
     }
 
-    @Synchronized
     @VisibleForTesting
     List<EventSegmentReader> getReaders() {
-        return Collections.unmodifiableList(readers);
+        synchronized (readers) {            
+            return Collections.unmodifiableList(readers);
```

Comment:
This is not really thread safe. `unmodifiableList` returns a wrapper for `readers`, but readers can still be modified by this object, thus causing issues for anyone who might be iterating over it at the same time.